Assignment4/Alfonsogm-160068/task07.py

```diff
@@ -0,0 +1,87 @@
+# -*- coding: utf-8 -*-
+"""Task07.ipynb
+
+Automatically generated by Colaboratory.
+
+Original file is located at
+    https://colab.research.google.com/drive/1wY-3ZSIbFRHEiVG-VZcWNWgN7ixQvrD_
+
+**Task 07: Querying RDF(s)**
+"""
+
+!pip install rdflib 
+github_storage = "https://raw.githubusercontent.com/FacultadInformatica-LinkedData/Curso2021-2022/master/Assignment4/course_materials"
+
+"""Leemos el fichero RDF de la forma que lo hemos venido haciendo"""
+
+from rdflib import Graph, Namespace, Literal
+from rdflib.namespace import RDF, RDFS
+g = Graph()
+g.namespace_manager.bind('ns', Namespace("http://somewhere#"), override=False)
+g.namespace_manager.bind('vcard', Namespace("http://www.w3.org/2001/vcard-rdf/3.0#"), override=False)
+g.parse(github_storage+"/rdf/example6.rdf", format="xml")
+
+for s, p, o in g:
+  print(s,p,o)
+
+"""**TASK 7.1: List all subclasses of "Person" with RDFLib and SPARQL**"""
+
+from rdflib.namespace import RDF, RDFS
+from rdflib.plugins.sparql import prepareQuery
+ns = Namespace("http://somewhere#")
+vcard = Namespace("http://www.w3.org/2001/vcard-rdf/3.0#")
+
+q1= prepareQuery('''
+Select ?subj
+Where {
+  ?subj rdfs:subClassOf ns:Person
+}
+''',
+  initNs = {"ns": ns,"rdfs":RDFS}
+)
+
+for r in g.query(q1):
+  print(r.subj)
+print("\n")
+for s, p, o in g.triples((None, RDFS.subClassOf, ns.Person)):
+  print(s)
+
+"""**TASK 7.2: List all individuals of "Person" with RDFLib and SPARQL (remember the subClasses)**
+
+"""
+
+q2= prepareQuery('''
+Select ?subj
+Where {
+  ?subj ?p ns:Person
+}
+''',
+  initNs = {"ns": ns,"rdfs":RDFS}
+)
+# Visualize the results
+for r in g.query(q2):
+  print(r.subj)
+print("\n")
+for s, p, o in g.triples((None, None, ns.Person)):
+  print(s)
+
+"""**TASK 7.3: List all individuals of "Person" and all their properties including their class with RDFLib and SPARQL**
+
+"""
+
+q3= prepareQuery('''
+Select ?subj ?p ?obj 
+Where {
+  ?subj ?p ?obj .
+  ?subj ?p1 ns:Person
```

Comment:
this is listing all the properties linked to Person, not all instances.
Also, you are missing subclasses. Incorrect